FireRoad/CourseSearchEngine.swift

```diff
@@ -85,6 +92,21 @@ struct SearchOptions: OptionSet {
         return true
     }
     
+    var whichSort: String {
```

Comment:
I see why you need to do this, but using strings feels dangerous :) can we define an enum for these sorting options? Maybe in the same file where you're going to put the sorting function. Then this `whichSort` property would instead be of the new sort options enum type.